src/main/native/CCM.c

```diff
@@ -501,14 +497,14 @@ JNIEXPORT jint JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_do_
         handleIV_CCM(ivLen, keyLen, blockSize, J0Offset, (char*) iv, (char*) (parm_block + keyOffset), (char*) parm_block);
         memcpy(parm_block + counterValueOffset, parm_block + J0Offset + blockSize - 4, 4); // Add Counter Value
 
-        CCMzS390(input, output, aad, &len, &alen, parm_block, &mode);
+        CCMzS390((input != NULL) ? (input + inputOffset) : NULL, (output != NULL) ? (output + outputOffset) : NULL, aad, &len, &alen, parm_block, &mode);
 
         // Copy tag
         tag = parameters + ivLen + aadLen + keyLen;
         memcpy(tag, parm_block + 16, tagLen); // Add tag to output
 
-        if(inputJ != NULL) (*env)->ReleasePrimitiveArrayCritical(env, inputJ, input, 0);
-        if(outputJ) (*env)->ReleasePrimitiveArrayCritical(env, outputJ, output, 0);
+        if(inputJ != NULL) (*env)->ReleasePrimitiveArrayCritical(env, inputJ, (input != NULL) ? (input + inputOffset) : NULL, 0);
```

Comment:
Why does the `ReleasePrimitiveArrayCritical` have different parameters than the `GetPrimitiveArrayCritical`? Shouldn't we just release `input`?

Also, should we check `if(input != NULL)` instead of `if(inputJ != NULL)`?

Reply:
I'm not sure what the difference is between the two methods.

There is a check for both `if(input != NULL)` and `if(inputJ != NULL)`.

Reply:
I mean that I think it should be 
`if(input != NULL) (*env)->ReleasePrimitiveArrayCritical(env, inputJ, input, 0);`
, since the get and release should be applied to the same arguments, not one with offset.

Reply:
Yes agreed i think we want to check if input is NULL then free `input` it seems more simple that way.